2D3D_UnityProject/Assets/Puzzles/Liars/LiarStatue.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class LiarStatue : MonoBehaviour
+{
+    /// <summary>
+    /// general statement this statue will say
+    /// </summary>
+    [SerializeField]
+    private string statement;
+
+    /// <summary>
+    /// what the statue will say when player looks for orb
+    /// </summary>
+    [SerializeField]
+    private string winLoseStatement;
+
+    /// <summary>
+    /// name of the statue
+    /// </summary>
+    [SerializeField]
+    private string statueName;
+
+    /// <summary>
+    /// subtitle of the statue
+    /// </summary>
+    [SerializeField]
+    private string subtitle;
+
+    /// <summary>
+    /// whether or not this statue has the orb
+    /// </summary>
+    [SerializeField]
+    private bool hasOrb;
+
+    /// <summary>
+    /// ID of this statue, used as index in manager list
+    /// </summary>
+    [SerializeField]
+    private int id;
+
+    void OnTriggerEnter(Collider other)
+    {
+        PlayerController controller = other.GetComponent<PlayerController>();
+        if (controller != null)
+            controller.SetInDialogueZone(true, id);
+        Debug.Log("Talking to " + statueName);
+    }
+
+    void OnTriggerExit(Collider other)
+    {
+        PlayerController controller = other.GetComponent<PlayerController>();
+        if (controller != null)
+            controller.SetInDialogueZone(false, id);
+        Debug.Log("Leaving conversation with " + statueName);
+    }
```

Comment:
This should probably be done on the player's end with a raycast in the direction player is facing. It could only cast on a NPC_Interact layer, so it would exclusively check for a dialogue source

Again this works for now and we can refactor later